mmdeploy/codebase/mmcls/models/backbones/vision_transformer.py

```diff
@@ -0,0 +1,254 @@
+                q_weight: Tensor, q_bias: Tensor, 
+                k_weight: Tensor, k_bias: Tensor, 
+                v_weight: Tensor, v_bias: Tensor,
+                o_weight: Tensor, o_bias: Tensor,
+                embed_dims: int, num_heads: int):
+
+        q_weight.setDebugName("q_weight")
+        q_bias.setDebugName("q_bias")
+
+        k_weight.setDebugName("k_weight")
+        k_bias.setDebugName("k_bias")
+
+        v_weight.setDebugName("v_weight")
+        v_bias.setDebugName("v_bias")
+
+        o_weight.setDebugName("o_weight")
+        o_bias.setDebugName("o_bias")
+        
+        return g.op(
+            'mmdeploy::MultiHeadAttention',
+            q, k, v,
+            q_weight, q_bias,
+            k_weight, k_bias,
+            v_weight, v_bias,
+            o_weight, o_bias,
+            embed_dim_i = embed_dims,
+            num_heads_i = num_heads)
+
+
+# ncnn have implemented MultiheadAttention, onnx would split this opr. So the model needs to rewrite.
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmcls.models.utils.attention.MultiheadAttention.forward',
+    backend=Backend.NCNN.value)
+def multiheadattention__forward__ncnn(ctx, self, qkv_input):
+    """Rewrite `forward` of MultiheadAttention used in vision_transformer for ncnn
+    backend.
+
+    Args:
+        ctx (ContextCaller): The context with additional information.
+        self (InvertedResidual): The instance of the class InvertedResidual.
+        x (Tensor): Input features of shape (N, Cin, H, W).
+    Returns:
+        out (Tensor): A feature map output from MultiHeadAttention. The tensor
+        shape (N, Cout, H, W).
+    """
+
+    # split qkv weight and bias
+    # [768, 2304] => 3 * [768, 768]
+    # qkv_weight = self.qkv.weight.data.reshape(self.input_dims, 3, self.embed_dims).permute(1, 0, 2)
+    qkv_weight = self.qkv.weight.data.reshape(3, self.input_dims, self.embed_dims)
+    
+    q_weight = qkv_weight[0]
+    k_weight = qkv_weight[1]
+    v_weight = qkv_weight[2]
+
+    # [2304] => 3 * [768]
+    qkv_bias = self.qkv.bias.data.reshape(3, self.embed_dims)
+    q_bias = qkv_bias[0]
+    k_bias = qkv_bias[1]
+    v_bias = qkv_bias[2]
+
+    # out weight and bias
+    o_weight = self.proj.weight.data
+    o_bias = self.proj.bias.data
+
+    out = MultiHeadAttentionop.apply(qkv_input, qkv_input, qkv_input, 
+        q_weight, q_bias, 
+        k_weight, k_bias, 
+        v_weight, v_bias, 
+        o_weight, o_bias, 
+        self.embed_dims, self.num_heads)
+    return out
+
+
+class GELUop(torch.autograd.Function):
+    """Create onnx::GELU op.
+    """
+
+    @staticmethod
+    def forward(ctx, inp: Tensor) -> Tensor:
+        return torch.rand_like(inp)
+
+
+    @staticmethod
+    def symbolic(g, inp: Tensor):
+        return g.op('mmdeploy::Gelu', inp)
+
+
+# ncnn have implemented MultiheadAttention, onnx would split this opr. So the model needs to rewrite.
+@FUNCTION_REWRITER.register_rewriter(

```

Comment:
As this is a rewrite of PyTorch function, it can be moved to [mmdeploy/pytorch](https://github.com/open-mmlab/mmdeploy/tree/master/mmdeploy/pytorch) so it can be used in other repos.
And what if you just want to rewrite the symbolic, `SYMBOLIC_REWRITER` would be a better choice. [here](https://github.com/open-mmlab/mmdeploy/tree/master/mmdeploy/pytorch/ops) are some examples.

Reply:
`GELU` and `LayerNorm` has been moved to `pytorch/ops` dir.